Sources/PackageRegistry/RegistryConfiguration.swift

```diff
@@ -76,16 +105,178 @@ extension RegistryConfiguration {
+            }()
+
+            public var onUnsigned: OnUnsignedAction?
+            public var onUntrustedCertificate: OnUntrustedCertificateAction?
+            public var trustedRootCertificatesPath: String?
+            public var includeDefaultTrustedRootCertificates: Bool?
+            public var validationChecks: ValidationChecks?
+
+            public init() {
+                self.onUnsigned = nil
+                self.onUntrustedCertificate = nil
+                self.trustedRootCertificatesPath = nil
+                self.includeDefaultTrustedRootCertificates = nil
+                self.validationChecks = nil
+            }
+
+            mutating func merge(_ other: Signing) {
+                if let onUnsigned = other.onUnsigned {
+                    self.onUnsigned = onUnsigned
+                }
+                if let onUntrustedCertificate = other.onUntrustedCertificate {
+                    self.onUntrustedCertificate = onUntrustedCertificate
+                }
+                if let trustedRootCertificatesPath = other.trustedRootCertificatesPath {
+                    self.trustedRootCertificatesPath = trustedRootCertificatesPath
+                }
+                if let includeDefaultTrustedRootCertificates = other.includeDefaultTrustedRootCertificates {
+                    self.includeDefaultTrustedRootCertificates = includeDefaultTrustedRootCertificates
+                }
+                if let validationChecks = other.validationChecks {
+                    self.validationChecks?.merge(validationChecks)
+                }
+            }
+
+            mutating func merge(_ other: ScopePackageOverride.Signing) {
+                if let trustedRootCertificatesPath = other.trustedRootCertificatesPath {
+                    self.trustedRootCertificatesPath = trustedRootCertificatesPath
+                }
+                if let includeDefaultTrustedRootCertificates = other.includeDefaultTrustedRootCertificates {
+                    self.includeDefaultTrustedRootCertificates = includeDefaultTrustedRootCertificates
+                }
+            }
+
+            public enum OnUnsignedAction: String, Hashable, Codable {
+                case error
+                case prompt
+                case warn
+                case silentAllow
+            }
+
+            public enum OnUntrustedCertificateAction: String, Hashable, Codable {
+                case error
+                case prompt
+                case warn
+                case silentTrust
+            }
+
+            public struct ValidationChecks: Hashable, Codable {
+                public var certificateExpiration: CertificateExpirationCheck?
+                public var certificateRevocation: CertificateRevocationCheck?
+
+                public init() {
+                    self.certificateExpiration = nil
+                    self.certificateRevocation = nil
+                }
+
+                mutating func merge(_ other: ValidationChecks) {
```

Comment:
this is more of an override than a merge?

Reply:
sure